hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala

```diff
@@ -383,6 +383,7 @@ class JSBuilder(using TL, State, Ctx) extends CodeBuilder:
           case Elaborator.ctx.builtins.Num => doc"typeof $sd === 'number'"
           case Elaborator.ctx.builtins.Bool => doc"typeof $sd === 'boolean'"
           case Elaborator.ctx.builtins.Int => doc"globalThis.Number.isInteger($sd)"
+          case Elaborator.ctx.builtins.TypedArray => doc"globalThis.ArrayBuffer.isView($sd) && !($sd instanceof globalThis.DataView)"
```

Comment:
> Even better: we can also add TypedArray to the Prelude and treat it as a virtual class

Do you mean something like this? (TypedArray is not an exposed type)

Reply:
Yes, that's exactly what I meant!